File: qc-application/src/main/java/com/eager/questioncloud/application/security/JwtAuthenticationManager.java
package com.eager.questioncloud.application.security;

import org.springframework.security.authentication.AuthenticationManager;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.AuthenticationException;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.stereotype.Component;

@Component
public class JwtAuthenticationManager implements AuthenticationManager {
    @Override
    public Authentication authenticate(Authentication authentication) throws AuthenticationException {
        SecurityContextHolder.getContext().setAuthentication(authentication);
        return authentication;
    }
}

File: qc-application/src/main/java/com/eager/questioncloud/application/security/JwtAuthenticationManager.kt
package com.eager.questioncloud.application.security

import org.springframework.security.authentication.AuthenticationManager
import org.springframework.security.core.Authentication
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.stereotype.Component

@Component
class JwtAuthenticationManager : AuthenticationManager {
    override fun authenticate(authentication: Authentication): Authentication {
        SecurityContextHolder.getContext().authentication = authentication
        return authentication
    }
}
